### DISCUSSION TITLE
How to set up encrypted user passwords when connecting Kafka engine tables to Kafka?

### DISCUSSION THREAD

**Discussion by hulxxx** [ClickHouse/ClickHouse#58852] [Q&A]
When connecting Kafka engine tables to Kafka, I can access a topic using a non-encrypted username and password. How do I connect using an encrypted password?

**Comment by UnamedRus** [comment]
What do you mean under encrypted?

You don't want to have it visible in table definition?

**Reply by hulxxx** [reply]
When using the Kafka engine table to connect to Kafka with SASL, you need to configure a username and password. Can encrypted passwords be used here?

**Reply by hulxxx** [reply]
I mean configuring the username and password in the config.xml

**Reply by UnamedRus** [reply]
Well there is https://github.com/ClickHouse/ClickHouse/issues/48291, but ClickHouse still need to decrypt them somehow.

Or you can store them in ENV, and just use tag     <password from_env="INTERPASSWORD"/>

https://github.com/ClickHouse/ClickHouse/pull/53818

https://kb.altinity.com/altinity-kb-integrations/altinity-kb-kafka/altinity-kb-adjusting-librdkafka-settings/#saslscram

**Reply by hulxxx** [reply]
This looks suitable for me. Thanks a lot.🙏
